### PR TITLE
Backend updated models

### PR DESCRIPTION
**Pull request by Kris2ffer**
Updated and created new models to prepare for model based interaction.
Some variables remains from old version so that it will still work with services, controllers and frontend.

[https://github.com/equinor/sepes/issues/111#issue-520954309](url)

### CODE REVIEW

**Review by Unicus-Tom — APPROVED** [review]

File: RestApi/Sepes.RestApi/Model/Pod.cs [hunk 1]
```diff
@@ -9,12 +10,38 @@ public class Pod
...
+            this.studyId = studyId;
+            this.allowAll = allowAll;
+            this.incoming = incoming.ToImmutableList();
+            this.outgoing = outgoing.ToImmutableList();
+            this.users = users.ToImmutableList();
+            this.locked = locked.ToImmutableList();
+            this.loaded = loaded.ToImmutableList();
+        }
 
         public Pod(ushort id, string name, int studyId)
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Model/Pod.cs]: Look into default argument or calling the big constructor with the small one using public Pod(ushort id, string name, int studyId): this(id, name, studyId, ImmutableList<Rule>.Empty, ....) {} pattern.

File: RestApi/Sepes.RestApi/Model/Pod.cs [hunk 2]
```diff
@@ -9,12 +10,38 @@ public class Pod
...
         {
             this.id = id;
             this.name = name;
             this.studyId = studyId;
+            this.allowAll = false;
+            this.incoming = new List<Rule>().ToImmutableList();
+            this.outgoing = new List<Rule>().ToImmutableList();
+            this.users = new List<User>().ToImmutableList();
+            this.locked = new List<DataSet>().ToImmutableList();
+            this.loaded = new List<DataSet>().ToImmutableList();
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Model/Pod.cs]: ImmutableList<Rule>.Empty is the Immutable way to create an empty list.

File: RestApi/Sepes.RestApi/Services/SepesDb.cs [hunk 3]
```diff
@@ -106,7 +106,7 @@ public async Task<int> createStudy(string studyName, int[] userIds, int[] datase
                 command.Parameters.AddWithValue("@podName", name);
                 command.Parameters.AddWithValue("@studyID", studyId);
                 var id = Convert.ToUInt16(await command.ExecuteScalarAsync());
-                return new Pod(id, name, studyId);
+                return new Pod(id, name, studyId, false, null, null, null, null, null);
```
Comment (gjersvik) [inline RestApi/Sepes.RestApi/Services/SepesDb.cs]: Do not like the use off null here. Null have a tendency blow up code that do not handle the null case explicitly.

**Review by gjersvik — CHANGES_REQUESTED** [review]
Look into using ImmutableList<T>.Empty and ImmutableHashSet<T>.Empty more.

**Review by gjersvik — APPROVED** [review]
Looks better.

**Pull request merged**